sql/core/benchmarks/ColumnarBatchBenchmark-jdk11-results.txt

```diff
@@ -0,0 +1,59 @@
+================================================================================================
+Int Read/Write
+================================================================================================
+
+OpenJDK 64-Bit Server VM 11.0.4+11-LTS on Linux 3.10.0-862.3.2.el7.x86_64
+Intel(R) Xeon(R) CPU E5-2670 v2 @ 2.50GHz
+Int Read/Write:                           Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
+------------------------------------------------------------------------------------------------------------------------
+Java Array                                          232            234           1       1409.7           0.7       1.0X
+ByteBuffer Unsafe                                   396            396           0        827.9           1.2       0.6X
+ByteBuffer API                                      602            603           0        544.0           1.8       0.4X
+DirectByteBuffer                                   1185           1186           2        276.6           3.6       0.2X
```

Comment:
For JDK8, `DirectByteBuffer` took `743 (Best)` and `744 (Avg)`. JDK11 seems to be affected due to Apache Spark's `Platform.java` implementation which depends on `java.version` for now. We need to fix or document this.

cc @srowen and @wangyum .

Reply:
Hm, I don't know. The cleaner hack that works in JDK 8 only works around an internal JVM check on max off heap limit. I would not imagine it causes slower performance, just may mean a user has to increase the off heap limit manually in extreme cases. That said, who knows what all the subtle implications are.

It could be some other cause too, of course, simply some difference in JDK behavior.

I don't think we want to force `--add-opens` or require people to use it of course, so I'd not benchmark that way. It's a valid observation in the benchmark, at least, to note that something seems to make it slower in JDK 11.

Reply:
Oh, got it. Thanks!